From: Nikolay Aleksandrov <nikolay@redhat.com>
Date: Tue, 9 Sep 2014 23:16:57 +0200
Subject: [PATCH 198/223] cxgb4: remove bond->lock
Patch-mainline: v3.18-rc1
Git-commit: 51752afa735ab28b6023fcf22ec63955107f8a65
References: bsc#909577 FATE#317550

RTNL should be already held in the notifier call so the slave list can
be traversed without a problem, remove the unnecessary bond->lock.

CC: Hariprasad S <hariprasad@chelsio.com>
Signed-off-by: Nikolay Aleksandrov <nikolay@redhat.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -4317,7 +4317,6 @@ static int cxgb4_inet6addr_handler(struc
 		 * bond. We need to find such different adapters and add clip
 		 * in all of them only once.
 		 */
-		read_lock(&bond->lock);
 		bond_for_each_slave(bond, slave, cnt) {
 			if (!first_pdev) {
 				ret = clip_add(slave->dev, ifa, event);
@@ -4331,7 +4330,6 @@ static int cxgb4_inet6addr_handler(struc
 				   to_pci_dev(slave->dev->dev.parent))
 					ret = clip_add(slave->dev, ifa, event);
 		}
-		read_unlock(&bond->lock);
 	} else
 		ret = clip_add(ifa->idev->dev, ifa, event);
 
